### DISCUSSION TITLE
cs_win_panel_round funfact_player data

### DISCUSSION THREAD

**Discussion by mavrick** [saul/demofile#224] [Questions] [answered]
Hey mate,

This is more a question. ``demo.gameEvents.on('cs_win_panel_round', e => { console.log(e.funfact_player); });`` returns a short http://wiki.sourcepython.com/developing/events/csgo.html#cs-win-panel-round

Any thoughts on how I can map this back to an actual player from ``demo.stringTables.findTableByName('userinfo')`` ?

**Comment by mavrick** [comment]
Hey mate,

Looks like the ``e.funfact_player`` returns the player entry index, not the ``userId``

I was able to get the player data using something like

```javascript
const getByUserIndex = (index) => {
  let userInfos = demo.stringTables.findTableByName('userinfo').entries;

  for (let i = 0; i < userInfos.length; ++i) {
    let userEntry = userInfos[i];

    if (userEntry.entry == index) {
      return userEntry.userData;
    }
  }

  return null;
}
```

Cheers for pointing me in the right direction :)

**Comment by saul** [comment, marked as answer]
You can use `demo.entities.entities.get(e.funfact_player)` to get the player entity. From there you can get the user info (`player.userInfo`) without having to manually iterate through the userinfo array

**Comment by zuloo** [comment]


**Reply by saul** [reply]
I've updated my comment again - the API has changed. You want `demo.entities.entities.get(e.funfact_player)` instead.

**Reply by zuloo** [reply]
yeah, that works. Thanks
